ci/scripts/python_test.sh

```diff
@@ -20,6 +20,7 @@
 set -ex
 
 arrow_dir=${1}
+test_dir=${1}/python/build/dist/temp
```

Comment:
Where is this "temp" directory coming from? Can we name it something else (e.g. "cpp_tests")?

Reply:
Hm ... not sure, can't remember. I think `ctest` selects it, but will check again and see if I can change it (if there is too much work I will leave it as is as we will be changing the tests for PyArrow cpp soon)